pandas/tests/frame/apply/test_frame_apply.py

```diff
@@ -179,6 +179,16 @@ def test_apply_with_string_funcs(self, request, float_frame, func, args, kwds, h
         expected = getattr(float_frame, func)(*args, **kwds)
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "how, args", [("pct_change", ()), ("nsmallest", (1, ["a", "b"])), ("tail", 1)]
+    )
```

Comment:
does this already raise when a function is passed? ideally can you co-locate these tests.

Reply:
When function is passed, apply with axis=1 computes the correct result. E.g.

```
df = DataFrame([[1, 2, 3], [4, 5, 6]])
print(df.apply(lambda x: x.tail(1), axis=1))

# gives:
   2
0  3
1  6
```

The argument `axis=1` is intercepted and then `apply_standard` is computed on the rows of the frame without an axis argument.

Reply:
ok cool, can you co-locate this test near the others

Reply:
What is meant by others? Currently located below the other test for str arguments. Tests for axis=1 and tests that raise are fairly spread out in this file, not sure how else to locate this test.

As an aside, I've been wanting to consolidate (and then start to cleanup) the apply/agg/transform tests into `pandas.tests.apply`. Does that move make sense?

Reply:
ok cool
yeah would love to break up some of these test files more logically